modules/ding_event/ding_event.admin.inc

```diff
@@ -25,17 +25,51 @@ function ding_event_admin_setting_form(array $form, array &$form_state) {
     '#description' => t("Select which currency to use on events"),
   );
 
-  $view = views_get_view('ding_event');
-  $view_items_per_page = $view->display['ding_event_list_frontpage']->display_options['pager']['options']['items_per_page'];
-  $default_items_per_page = variable_get('ding_event_frontpage_items_per_page', $view_items_per_page);
-
-  $form['ding_event']['ding_event_frontpage_items_per_page'] = array(
-    '#type' => 'textfield',
-    '#title' => t('Maximum number of events on front page.'),
-    '#size' => 4,
-    '#description' => t('Maximum number of events on front page.'),
-    '#default_value' => $default_items_per_page,
-    '#element_validate' => array('element_validate_integer_positive'),
+  $form['ding_event_frontpage'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Frontpage'),
+  );
+  $form['ding_event_frontpage']['ding_event_frontpage_items_per_page'] = array(
+    '#type' => 'select',
+    '#title' => t('Events in frontpage list'),
+    '#options' => array(
+      '6' => '3 - 6',
+      '12' => '6 - 12',
+      '18' => '9 - 18',
```

Comment:
If an event has an image it takes up the space of two event. Therefore the view will show 9 events if they all have images and 18 if none have images.